packages/serverpod/lib/src/database/database_result.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:serverpod/database.dart';
+
+/// Prepares a query result for serverpod serialization.
```

Comment:
Perhaps we should declare this `@Internal`? Could be other places too. Plus explain in docs that this isn't usually called by our users.

Reply:
I was conflicted if I should use the `@internal` annotation. It was great for debugging purposes to ensure that we didn't expose it as part of exported libraries but after that I removed it from most places.

Feels like we need to have a policy around how these tags should be used if we introduce them?
I'll add some docs in the right places that explain these things should not be called by our users.

Reply:
Added the `@internal` annotation to the new classes.